uibackend/server/dashboard_riskiest_assets.go

```diff
@@ -111,9 +111,25 @@ func (s *ServerImpl) getRiskiestAssetsForFindingType(ctx context.Context, findin
 }
 
 func (s *ServerImpl) getRiskiestAssetsForVulnerabilityType(ctx context.Context) ([]types.VulnerabilityRiskyAsset, error) {
-	assets, err := s.getRiskiestAssetsPerFinding(ctx, apitypes.VULNERABILITY)
+	totalFindingField, err := getTotalFindingFieldName(apitypes.VULNERABILITY)
 	if err != nil {
-		return nil, fmt.Errorf("failed to get riskiest assets: %w", err)
+		return nil, fmt.Errorf("failed to get total findings field name: %w", err)
+	}
+
+	totalVulnerabilitiesNotZeroFilter := ""
+	for _, field := range orderedSeveritiesFields {
+		totalVulnerabilitiesNotZeroFilter += fmt.Sprintf("summary/%s.%s gt 0 or ", totalFindingField, field)
+	}
+	totalVulnerabilitiesNotZeroFilter = strings.TrimSuffix(totalVulnerabilitiesNotZeroFilter, " or ")
```

Comment:
makes it a bit clearer but not a real substitute for the fact that we need a proper query builder 😄 
```suggestion
	vulnerabilityNotZeroFields := make([]string, len(orderedSeveritiesFields))
	for i, field := range orderedSeveritiesFields {
		vulnerabilityNotZeroFields[i] = fmt.Sprintf("summary/%s.%s gt 0", totalFindingField, field)
	}
	totalVulnerabilitiesNotZeroFilter := strings.Join(vulnerabilityNotZeroFields, " or ")
```